### ISSUE TITLE
listen_state should take into account entity's `last_changed` for immediate triggers

### ISSUE THREAD

**Issue by Ten0** [AppDaemon/appdaemon#2186]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Your feature request

Currently, for `immediate` `listen_state`s, AppDaemon will wait at least the specified `duration` after AppDaemon's start before running the trigger, even if we can know that we can actually wait for less time and still be sure that the state we are listening for has not changed for longer than that time.

We could look at entity's `last_changed` that comes back from HomeAssistant's API to solve this.
This would make such triggers survive AppDaemon reboots/reloads much better, as it would mean that such triggers could typically recover their "I've already waited for this long" state.

**Comment by runningman84** [comment]
My current approach is to have two listeners one for immediate and another with the wait duration. In the callback function I check the last updated attribute and decide what to do…

**Comment by Ten0** [comment]
Yeah that works but it's pretty verbose and seems like something that would have its place in AppDaemon's framework.
